src/Store.cs

```diff
@@ -138,6 +138,12 @@ public interface IStore
         /// <summary>
         /// Gets the context of the store.
         /// </summary>
+        /// <remarks>
+        /// Note: Generally, you must keep the <see cref="IStore"/> alive (by using
+        /// <see cref="GC.KeepAlive(object)"/>) until the <see cref="StoreContext"/> is no longer
+        /// used, to prevent the the Store.Handle finalizer from prematurely deleting the handle 
+        /// in the GC finalizer thread while the <see cref="StoreContext"/> is still in use.
+        /// </remarks>
         StoreContext Context { get; }
```

Comment:
Thanks for adding this comment about safety!

As I mentioned in #200, I think it might be for the best that we eliminate `IStore` since it exposes this unsafe behavior unnecessarily. If we do, we should definitely keep this comment around such that wherever we construct a `StoreContext` the caller must know to keep the store alive for the duration of it being used.